x-pack/platform/plugins/shared/streams/server/routes/streams/significant_events/route.ts

```diff
@@ -158,7 +167,79 @@ const readSignificantEventsRoute = createServerRoute({
   },
 });
 
+const generateSignificantEventsRoute = createServerRoute({
+  endpoint: 'GET /api/streams/{name}/significant_events/_generate 2023-10-31',
+  params: z.object({
+    path: z.object({ name: z.string() }),
+    query: z.object({
+      connectorId: z.string(),
+      currentDate: dateFromString.optional(),
+    }),
+  }),
+
+  options: {
+    access: 'public',
+    summary: 'Generate significant events',
+    description: 'Generate significant events queries based on the stream data',
+    availability: {
+      since: '9.2.0',
+      stability: 'experimental',
+    },
+  },
+  security: {
+    authz: {
+      requiredPrivileges: [STREAMS_API_PRIVILEGES.read],
+    },
+  },
+  handler: async ({
+    params,
+    request,
+    getScopedClients,
+    server,
+    logger,
+  }): Promise<
+    Observable<ServerSentEventBase<'generated_queries', { query: GeneratedSignificantEventQuery }>>
+  > => {
+    const isAvailableForTier = server.core.pricing.isFeatureAvailable(
+      STREAMS_TIERED_SIGNIFICANT_EVENT_FEATURE.id
+    );
+    if (!isAvailableForTier) {
+      throw new SecurityError(`Cannot access API on the current pricing tier`);
+    }
+
+    const { streamsClient, scopedClusterClient, licensing, inferenceClient } =
+      await getScopedClients({ request });
+
+    await assertEnterpriseLicense(licensing);
+    const isStreamEnabled = await streamsClient.isStreamsEnabled();
+    if (!isStreamEnabled) {
+      throw badRequest('Streams are not enabled');
+    }
+
+    const generatedSignificantEventDefinitions = await generateSignificantEventDefinitions({
```

Comment:
FWIW, this blocks the stream until the operation has been resolved. So I think there's a risk of a timeout here.